crates/runtime/src/datafusion.rs

```diff
@@ -297,7 +300,13 @@ impl DataFusion {
                         .context(UnableToCreateMemoryStreamSnafu)?,
                     );
 
-                    return Ok(QueryResult::new(record_batch_stream, Some(true)));
+                    return Ok(QueryResult::new(
+                        attach_query_context_to_stream(
```

Comment:
We can potentially finish context here as we know number of records and schema and don't wrap cached response with context to finish it once stream is completed but benchmarking does not show any performance gain so keeping unified logic here.